### PR TITLE
TLI 101.f -- Refactored Demographics for Clean Architecure

### PR DESCRIPTION
**Pull request by Jai0212**
#### Description
This PR refactors the demographics section of the `dashboard.jsx` file into smaller, reusable components to enhance readability, maintainability, and clarity. By separating the demographics-related code into distinct components, we aim to achieve better organization and ensure the codebase remains easy to document and maintain in the future.

#### Changes
- **Componentization**:
  - Extracted demographics-related logic and UI into multiple smaller components.
  - Each component now handles a specific part of the demographics section.
- **Refactoring**:
  - Updated `dashboard.jsx` to import and use the new components.
  - Cleaned up state management and props passing to ensure seamless communication between components.
- **Functionality**:
  - Ensured the demographic selection on **Dashboard 1** continues to function as expected.
  - Verified that demographic values are passed correctly between pages.

#### Acceptance Criteria
- [x] Demographic selection on **Dashboard 1** remains functional.
- [x] Correct demographic values are passed between pages.
- [x] Code is refactored into smaller, reusable components.
- [x] State management and props passing work as intended.
- [x] All relevant functionality is tested and validated after the refactor.
- [x] Documentation is updated to reflect the changes.

#### Testing
- Manually tested the demographic selection functionality on **Dashboard 1**.
- Verified values are correctly passed and reflected on subsequent pages.
- Confirmed no regression in other parts of the dashboard.

#### Documentation
Updated documentation to include details about the new components, their usage, and any relevant props or state requirements.

#### Impact
This refactor improves the overall maintainability of the codebase and lays the foundation for easier feature additions or modifications in the demographics section.

#### Reviewers
Please focus on:
1. Code structure and componentization.
2. Correctness of state management and props passing.
3. Functional testing to ensure no regressions.

### CODE REVIEW

File: frontend/src/pages/DashboardPage/ChartComponenet/ChartComponent.jsx [hunk 1]
```diff
@@ -145,7 +145,7 @@ const ChartComponent = forwardRef(({ chartData, sliderValue }, ref) => {
               label: (tooltipItem) => {
                 const item = accuracyData[tooltipItem.dataIndex];
                 return [
-                  `Accuracy: ${item.accuracy}`,
+                  `Bias: ${1 - item.accuracy}`,
```
Comment (Navnoorsinghmahal) [inline frontend/src/pages/DashboardPage/ChartComponenet/ChartComponent.jsx:148]: Did you check with Akshat about the label of y_axis

File: frontend/src/pages/DashboardPage/Demographics/DemographicsSelector.jsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
+import React from "react";
+
+const DemographicsSelector = ({
```
Comment (Navnoorsinghmahal) [inline frontend/src/pages/DashboardPage/Demographics/DemographicsSelector.jsx:3]: Is the dashboard functional with the changes you made?

**Review by Navnoorsinghmahal — COMMENTED** [review]
Reviewed and the code is fine.

File: frontend/src/pages/DashboardPage/ChartComponenet/ChartComponent.jsx (hunk 1 above)
Comment (Jai0212) [reply frontend/src/pages/DashboardPage/ChartComponenet/ChartComponent.jsx:148]: Yes I checked with him

Comment (armaganngul) [inline frontend/src/pages/DashboardPage/ChartComponenet/ChartComponent.jsx:148]: We should probably add this in the info button?

**Review by armaganngul — COMMENTED** [review]
Looks good!

File: frontend/src/pages/DashboardPage/Demographics/DemographicsSelector.jsx (hunk 2 above)
Comment (Jai0212) [reply frontend/src/pages/DashboardPage/Demographics/DemographicsSelector.jsx:3]: Yea its working

File: frontend/src/pages/DashboardPage/ChartComponenet/ChartComponent.jsx (hunk 1 above)
Comment (Jai0212) [reply frontend/src/pages/DashboardPage/ChartComponenet/ChartComponent.jsx:148]: Thats a good catch. We need to work on that

**Pull request merged**